install/helm/open-match/templates/backendapi.yaml

```diff
@@ -42,20 +39,15 @@ metadata:
   labels:
     app: {{ template "openmatch.name" . }}
     component: backend
-    chart: {{ template "openmatch.chart" . }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+    {{- include "openmatch.chartmeta" (set . "indent" 4) }}
 spec:
   replicas: 1
   selector:
     matchLabels:
       app: {{ template "openmatch.name" . }}
       component: backend
-      chart: {{ template "openmatch.chart" . }}
-      release: {{ .Release.Name }}
-      heritage: {{ .Release.Service }}
+      {{- include "openmatch.chartmeta" (set . "indent" 6) }}
   template:
-    namespace: {{ .Release.Namespace }}
```

Comment:
What namespace is this going into? Or was this unncessary?

Reply:
Pod will be created in the namespace that is specified at [`.metadata.namespace` of Deployment definition](https://github.com/GoogleCloudPlatform/open-match/pull/119/files/12a44bcf2b6e6ee12e269388dc460a432b873b09#diff-ff3081741227225fc85e625cbc4c142aR38). That was not valid k8s config actually, kubectl complained about that.

Reply:
BTW it seems to me that it's better to have namespace-agnostic k8s configs and let anybody pass the desired namespace with "-n" arg.
- People may want to install Open-match in any namespace they want, and we'd better not make them do replaces in the file.
- The YAML output that Redis chart produces does not contain a namespace in resources definitions, so the components will be installed in "default" unless other is specified with `kubectl --namespace XXX apply -f ...`. On the other hand Open-match components expect to find Redis service by its name that currently is automatically substituted by Kubernetes (`$(OM_REDIS_MASTER_SERVICE_HOST)`) - in the same namespace. Of course it's possible to make Open-match talk to Redis in different namespace but again not without of editing k8s files.

Reply:
So I've removed "namespace" from Open-match k8s YAML. However one thing that is still coupled to a hardcoded namespace is a Service Accounts: "mmf-sa" in case of Open-match, and similar ones for Prometheus and Grafana. Not sure about how to deal with it...

Reply:
I'm fine having a namespace as a parameter but using default is not desirable.

Reply:
You can install resources in any namespace with `kubectl --namespace XXX apply -f file.yaml` if it's not specified. In contrast if namespace is set explicitly in YAML then kubectl will complain in that case.